bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueGroupHandler.java

```diff
@@ -397,6 +418,33 @@ public void onGroupGone() {
         updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.GONE, "@text/offline.group-removed");
     }
 
+    /**
+     * Sets the state options for applicable scenes.
+     */
+    @Override
+    public void onScenesUpdated(List<Scene> updatedScenes) {
+        List<StateOption> stateOptions = Collections.emptyList();
+        consoleScenesList = new ArrayList<>();
+        HueClient handler = getHueClient();
+        if (handler != null) {
+            FullGroup group = handler.getGroupById(groupId);
+            if (group != null) {
+                stateOptions = updatedScenes.parallelStream()//
```

Comment:
It was @leluna initial choice.
Yes, I think it will be generally less than 100, except if you have defined a very big number of rooms in your home or you created a lot of hue scenes.

Reply:
Done